src/common/s3_adapter.cpp

```diff
@@ -122,6 +122,29 @@ void S3Adapter::Deinit() {
     delete throttle_;
 }
 
+void S3Adapter::Reinit(const std::string& ak, const std::string& sk,
+                       const std::string& endpoint) {
+    Deinit();
```

Comment:
`Deinit` will call `Aws::ShutdownAPI`, it's a global function and may affect all s3adaptors, you'd better check this

Reply:
yes, current Init() and Deinit() will cause problem, AWS InitAPI and ShutdownAPI should be called only once per program.

Reply:
will make changes in another PR first.

Reply:
pull #678